packages/@ember/-internals/metal/lib/computed.ts

```diff
@@ -675,6 +676,25 @@ export class ComputedProperty extends ComputedDescriptor {
     let ret;
 
     try {
+      if (obj instanceof Ember.Component) {
+        // ensure two way binding works when the component has defined a computed
+        // property with both a setter and dependent keys, in that scenario without
+        // the sync observer added below the caller's value will never be updated
+        //
+        // See GH#18417 / GH#19028 for details.
+        let descriptor: ComputedDescriptor;
+        descriptor = descriptorForProperty(obj, keyName);
+        let meta = metaFor(obj);
```

Comment:
Since `descriptorForProperty` accepts an optional meta (and we are just going to look one up immediately after calling) we should do the meta lookup first and pass it to `descriptorForProperty`:

```suggestion
        let meta = metaFor(obj);
        let descriptor = descriptorForProperty(obj, keyName, meta);
```